account_asset_management/models/account_asset.py

```diff
@@ -0,0 +1,1103 @@
+                        fy_amount = fy_residual_amount - self.salvage_value
+                    else:
+                        fy_amount = fy_residual_amount
+                else:
+                    firstyear = i == 0 and True or False
+                    fy_factor = self._get_fy_duration_factor(
+                        entry, firstyear)
+                    fy_amount = year_amount * fy_factor
+                if asset_sign * (fy_amount - fy_residual_amount) > 0:
+                    fy_amount = fy_residual_amount
+                period_amount = round(period_amount, digits)
+                fy_amount = round(fy_amount, digits)
+            else:
+                fy_amount = False
+                if self.method_time == 'number':
+                    number = self.method_number
+                elif self.method_time == 'end':
+                    number = len(line_dates)
+                period_amount = round(self.depreciation_base / number, digits)
+
+            entry.update({
+                'period_amount': period_amount,
+                'fy_amount': fy_amount,
+            })
+            if self.method_time == 'year':
+                fy_residual_amount -= fy_amount
+                if round(fy_residual_amount, digits) == 0:
+                    break
+        i_max = i
+        table = table[:i_max + 1]
+
+        # Step 2:
+        # Spread depreciation amount per fiscal year
+        # over the depreciation periods.
+        self._compute_depreciation_table_lines(
+            table, depreciation_start_date, depreciation_stop_date,
+            line_dates)
+
+        return table
+
+    def _get_depreciation_entry_name(self, seq):
+        """ use this method to customise the name of the accounting entry """
+        return (self.code or str(self.id)) + '/' + str(seq)
+
+    @api.multi
+    def _compute_entries(self, date_end, check_triggers=False):
+        # To DO : add ir_cron job calling this method to
+        # generate periodical accounting entries
+        result = []
+        error_log = ''
+        if check_triggers:
+            recompute_obj = self.env['account.asset.recompute.trigger']
+            recomputes = recompute_obj.sudo().search([('state', '=', 'open')])
+        if check_triggers and recomputes:
+            trigger_companies = recomputes.mapped('company_id')
+            for asset in self:
+                if asset.company_id.id in trigger_companies.ids:
+                    asset.compute_depreciation_board()
+
+        depreciations = self.env['account.asset.line'].search([
+            ('asset_id', 'in', self.ids),
+            ('type', '=', 'depreciate'),
+            ('init_entry', '=', False),
+            ('line_date', '<=', date_end),
+            ('move_check', '=', False)],
+            order='line_date')
+        for depreciation in depreciations:
+            try:
+                with self.env.cr.savepoint():
+                    result += depreciation.create_move()
+            except:
```

Comment:
I am not sure if it would be appropriate to include an empty `except` statement. If we expect exceptions then we should know what kind of exceptions do we expect and how to recover from them